packages/2d/src/lib/components/Code.ts

```diff
@@ -368,7 +376,123 @@ export class Code extends Shape {
    * @param pattern - Either a string or a regular expression to match.
    */
   public findLastRange(pattern: string | RegExp): CodeRange {
-    return findAllCodeRanges(this.parsed(), pattern).at(-1) ?? [[0, 0], [0, 0]];
+    return (
+      findAllCodeRanges(this.parsed(), pattern).at(-1) ?? [
```

Comment:
Is there optimization we should worry about here? Normally `findLast` or `findFirst` are used to avoid a full string iteration if possible, although I'm not sure it's worth it here.

Reply:
I don't know how to do that while supporting regex tbh.
The function is mainly for convenience but I'm always open to optimization propositions.